syslibs/opcua/CMakeLists.txt

```diff
@@ -1,15 +1,22 @@
-
 add_ov_library(opcua)
 ov_library_includes(opcua ov ksbase)
 
+#alternative syntax:
+set(UA_NAMESPACE_ZERO "REDUCED" CACHE STRING "Enable reduced NS0 (default) to allow for faster server startup." FORCE)
+set(UA_ENABLE_AMALGAMATION "ON" CACHE BOOL "Merge als .c and .h files from open62541 in one open62541.c/.h" FORCE)
+set(UA_AMALGAMATION_ARCHITECTURES "posix;win32" CACHE STRING "Include posix and win32 architecture." FORCE)
+set(UA_ENABLE_IMMUTABLE_NODES "ON" CACHE BOOL "Use getNodeCopy and replaceNode instead of modifying nodes directly." FORCE)
+set(UA_ENABLE_NODEMANAGEMENT "OFF" CACHE BOOL "Nodemanagement is currently not implemented, as no backtransformations are implemented." FORCE)
+add_subdirectory(open62541)
```

Comment:
A nice feature here would be defining a custom CMake option to switch between including the open62541 sources from the subdirectory and linking to a system-provided pre-build open62541 by using `find_package(open62541 ...)` as explained in https://open62541.org/doc/current/installing.html#manual-installation.

Additionally, we should make sure, open62541 is only built/searched when required. We could add an additional `OV_WITH_OPCUA` option, and bascially wrap this full CMakeLists file into an `if(OV_WITH_OPCUA) … endif()`. When disabling this option, open62541 will not be included/searched, but the opcua OV library will not be available as well.

Reply:
> I don't see the need for an OV_WITH_OPCUA option, as the library can be switched ov via SET(BUILD_OV_LIBRARY_opcua OFF CACHE BOOL "") analogous to kshist if desired.

However, if you set `BUILD_OV_LIBRARY_opcua` to off, open62541 will still be included into the CMake project. This may result in unnecessarily building open62541 and may lead to further problems when open62541 is not available (neither as submodule nor as a systemwide-installation). It may also result in problems when building OV for a platform that is not not supported by open62541. In those situations, it would be nice to have a switch, that disables including open62541 all together.

Reply:
Ok, so can we just reuse the `BUILD_OV_LIBRARY_opcua` option, as shown in 6bab542.

Reply:
Hm. Originally, the `BUILD_OV_LIBRARY_*` has other semantics: It controls whether the library is added to CMake's default build target, i.e. it is build when simply running `make` (or whatever build system you are using) without a specific target. When you disable the default build using the relevant option, it should (by now) still be possible to build the library explicitly (`make opcua`) or as a dependency of another library.

This is why I propose to add another flag (something like `OV_WITH_OPCUA`) to completely disable opcua/open62541 support.